### ISSUE TITLE
celery beat not working with flask-socketio

### ISSUE THREAD

**Issue by deounix** [miguelgrinberg/Flask-SocketIO#608]
The point of what I'm doing is run async function every 1 sec that get data from some site and insert this data into my database.
I'm working with windows env.

my code:

```
# celery
from celery import Celery

def make_celery(app):
	celery = Celery(app.import_name, broker=app.config["CELERY_BROKER_URL"])
	celery.conf.update(app.config)
	TaskBase = celery.Task
	class ContextTask(TaskBase):
		abstract = True
		def __call__(self, *args, **kwargs):
			with app.app_context():
				return TaskBase.__call__(self, *args, **kwargs)
	celery.Task = ContextTask
	return celery

celery = make_celery(app)


@celery.on_after_configure.connect
def setup_periodic_tasks(sender, **kwargs):
    # Calls test('hello') every 10 seconds.
    sender.add_periodic_task(10.0, test.s('hello'), name='add every 10')

@celery.task
def test(arg):
    print(arg)
```

Flask-socketio with eventlet async_mode + celery I'm able to run it like this (am using it for mail func) : 

`celery -A app.celery worker -P eventlet`

Then I'm trying to run celery beat but I can't

`celery -A app.celery beat`

error : 

```
    raise NotImplementedError("set_nonblocking() on a file object "
NotImplementedError: set_nonblocking() on a file object with no setblocking() method (Windows pipes don't support non-blocking I/O)
```

and celery beat doesn't have -P option.

What should I do?

**Comment by miguelgrinberg** [comment]
Do you need to use eventlet in the Celery worker? Flask-SocketIO does not need to be async in the external processes, just the server needs async.

**Comment by deounix** [comment]
if am trying to run celery with out eventlet it won't run cuz Flask-SocketIO using eventlet in async mode

eg: celery -A app.celery worker (not working)

What should I do?

**Comment by miguelgrinberg** [comment]
No, wait. You have two processes, right? The Flask server runs on eventlet, so you create a `SocketIO` instance which is initialized to work with eventlet. Your second process is the Celery worker. When you create the `SocketIO` object for the Celery side, you don't have to use eventlet, you can create it by passing `async_mode='threading'` as an argument and that will skip the async stuff. That's how I do it, see https://github.com/miguelgrinberg/flack/blob/master/flack/__init__.py#L39-L52 for an example.

**Comment by deounix** [comment]
My project look like this:

""
app= Flask(__name__)
socketio = socketio(app, async_mode='eventlet')
""

here I'm not using gunicorn yet cuz I'm stl in dev case

after that am trying to use celery within the main file but I can't do this with out -P eventlet. (cuz socketio already use it in async mode) and celery must use it also that's because (eventlet.monkey_patch) won't let celery run without -P eventlet

flask app --> socketio with eventlet + monkey patch --> celery + eventlet to work 100%

In ur code u switch to threading and let socketio use message broker so is this good for production mode?

**Comment by deounix** [comment]
Hi @miguelgrinberg,

This repo explain what I'm doing : https://github.com/deounix/flask-socketio-celery

thnx.

**Comment by miguelgrinberg** [comment]
As I said above, you don't need eventlet for the Celery worker. Just create your app using the threading mode for SocketIO.
